statistics/selectivity.go

```diff
@@ -44,32 +44,66 @@ const (
 	colType
 )
 
-// checkColumnConstant receives two expressions and makes sure one of them is column and another is constant.
-func checkColumnConstant(e []expression.Expression) bool {
+// getConstantColumnID receives two expressions and if one of them is column and another is constant, it returns the
+// ID of the column.
+func getConstantColumnID(e []expression.Expression) int64 {
 	if len(e) != 2 {
-		return false
+		return -1
 	}
-	_, ok1 := e[0].(*expression.Column)
+	col, ok1 := e[0].(*expression.Column)
 	_, ok2 := e[1].(*expression.Constant)
 	if ok1 && ok2 {
-		return true
+		return col.ID
 	}
-	_, ok1 = e[1].(*expression.Column)
+	col, ok1 = e[1].(*expression.Column)
 	_, ok2 = e[0].(*expression.Constant)
-	return ok1 && ok2
+	if ok1 && ok2 {
+		return col.ID
+	}
+	return -1
 }
 
-func pseudoSelectivity(exprs []expression.Expression) float64 {
+func pseudoSelectivity(t *Table, exprs []expression.Expression) float64 {
 	minFactor := selectionFactor
+	uniqueCol := make(map[string]bool)
 	for _, expr := range exprs {
-		if fun, ok := expr.(*expression.ScalarFunction); ok && checkColumnConstant(fun.GetArgs()) {
-			switch fun.FuncName.L {
-			case ast.EQ, ast.NullEQ:
-				minFactor = math.Min(minFactor, 1.0/pseudoEqualRate)
-			case ast.GE, ast.GT, ast.LE, ast.LT:
-				minFactor = math.Min(minFactor, 1.0/pseudoLessRate)
-				// FIXME: To resolve the between case.
+		fun, ok := expr.(*expression.ScalarFunction)
+		if !ok {
+			continue
+		}
+		colID := getConstantColumnID(fun.GetArgs())
+		if colID == -1 {
+			continue
+		}
+		switch fun.FuncName.L {
+		case ast.EQ, ast.NullEQ, ast.In:
```

Comment:
`In` and `EQ` has the same selectivity?

Reply:
`getConstantColumnID` guarantees there exists only two parameters for the current expr, so `in` is equivalent to `eq`